src/orderly.py

```diff
@@ -1,13 +1,103 @@
 from subprocess import run
 
+import os
+import os.path
+import paths
+
 import versions
-from docker_helpers import get_image_name
+from docker_helpers import get_image_name, docker_cp
+from service import orderly_volume_name
+
+from settings import get_secret, save_secret
 from service import orderly_volume_name
 
+orderly_ssh_keypath = ""
 
-def create_empty_store():
-    print("Creating empty Orderly store")
-    image = get_image_name("montagu-reports", versions.reports)
+def create_orderly_store(settings):
+    print("Creating orderly store")
+    image = get_image_name("montagu-orderly", versions.reports)
     run(["docker", "pull", image], check=True)
-    template = "docker run --rm --entrypoint /usr/bin/orderly_init -v {volume}:/orderly {image} /orderly"
-    run(template.format(volume=orderly_volume_name, image=image).split(' '), check=True)
+    if settings["initial_data_source"] == "restore":
+        restore_orderly_store()
+    else:
+        configure_orderly_store(settings)
+
+def restore_orderly_store():
+    print("Restoring orderly permissions")
+    args = ["docker", "run", "--rm", "-d",
+            "--entrypoint", "chmod",
+            "--name", "orderly_setup",
+            "-v", orderly_volume_name + ":/orderly",
+            get_image_name("montagu-orderly", versions.reports),
+            "600", "/orderly/.ssh/id_rsa"]
+    run(args)
+
+def configure_orderly_store(settings):
+    clone = settings['clone_reports']
+    use_real_passwords = settings['use_real_passwords']
+
+    ssh = orderly_prepare_ssh(clone)
+    envir = orderly_prepare_envir(use_real_passwords)
+
+    container = "orderly_setup"
+    args = ["docker", "run", "--rm", "-d",
+            "--entrypoint", "sleep",
+            "--name", container,
+            "-v", orderly_volume_name + ":/orderly",
+            get_image_name("montagu-orderly", versions.reports),
+            "3600"]
```

Comment:
Why 3600?

Reply:
"long enough" (1 hour).  This is a complication of not having a process running but wanting to get things into the container - I'm running `sleep 3600` to keep the container up and then stopping it later.  The number won't affect things unless it's too short

Reply:
I've added a note in cb3b791

Reply:
Why not `sleep infinity` then? If you are explicitly killing the container anyway.

Reply:
TIL `sleep infinity`

Reply:
(This is implemented now - it's apparently linux specific but as it only runs in a container that we know has a suitable bash/linux that's ok)